### ISSUE TITLE
"too big" messages get dropped silently

### ISSUE THREAD

**Issue by ctron** [knative-sandbox/eventing-kafka#279]
**Describe the bug**

A message that is considered "too big" is getting dropped silently. Only in the logs you see the following message:

~~~
{"level":"warn","ts":"2020-12-15T17:08:21.169Z","logger":"kafkachannel-dispatcher","caller":"dispatcher/dispatcher.go:278","msg":"Got","error":"kafka: Failed to produce message to topic knative-messaging-kafka.drogue-iot.iot-channel: kafka server: Message was too large, server rejected it to avoid allocation error."}
~~~

The HTTP request to the sink still reports:

~~~
[2020-12-15T17:42:44Z DEBUG drogue_cloud_endpoint_common::downstream] Content-Type: Some("audio/vnd.wave;codec=1")
[2020-12-15T17:42:44Z DEBUG drogue_cloud_endpoint_common::downstream] Payload size: 1753132 bytes
[2020-12-15T17:42:44Z DEBUG hyper::client::pool] reuse idle connection for ("http", iot-channel-kn-channel.drogue-iot.svc.cluster.local) 
[2020-12-15T17:42:44Z DEBUG hyper::client::pool] pooling idle connection for ("http", iot-channel-kn-channel.drogue-iot.svc.cluster.local) 
[2020-12-15T17:42:44Z DEBUG reqwest::async_impl::client] response '202 Accepted' for http://iot-channel-kn-channel.drogue-iot.svc.cluster.local/
~~~

 **Expected behavior**

Either the message should be accepted, or the return value of the sink should indicate an error.

 **To Reproduce**

Post a message that is "too big" to a `KafkaChannel`.

 **Knative release version**

0.19.1

 **Additional context**
Add any other context about the problem here such as proposed priority

**Comment by slinkydeveloper** [comment]
On one hand, I should point out that CloudEvents payloads are meant to carry small data payloads https://github.com/cloudevents/spec/blob/v1.0.1/spec.md#size-limits

Given that, this error is weird, because the message should be rejected... Did you checked your Kafka config for some "maximum record size" conf or something like that?

**Comment by ctron** [comment]
> On one hand, I should point out that CloudEvents payloads are meant to carry small data payloads https://github.com/cloudevents/spec/blob/v1.0.1/spec.md#size-limits

True, however: if a message gets rejected, then it shouln't return `202`.

The kafka channel is pretty standard:

~~~yaml
apiVersion: messaging.knative.dev/v1alpha1
kind: KafkaChannel
metadata:
  name: iot-channel
spec:
  numPartitions: 3
  replicationFactor: 1
~~~

**Comment by slinkydeveloper** [comment]
> The kafka channel is pretty standard:

I'm talking about the actual kafka instance used by the KafkaChannel, does it has a record limit?

**Comment by ctron** [comment]
> > The kafka channel is pretty standard:
> 
> I'm talking about the actual kafka instance used by the KafkaChannel, does it has a record limit?
> 

No special settings.

> Also, are you using the latest kafka channel [0.19.3](https://github.com/knative-sandbox/eventing-kafka/releases/download/v0.19.3/channel-consolidated.yaml)?

We using 0.19.1.

> 
> I feel like you're using an old version that doesn't contain this fix #181

From the commit of this PR I cannot see in which version this landed. So is 0.19.1 too old?

**Comment by lionelvillard** [comment]
@ctron yes 0.19.1 is too old for your particular use case. The fix landed in 0.19.3, which was cut 3 days ago. 

We could definitively do better with the release notes. Today is really hard to know what is fixed in patch releases. Pinging @n3wscott for help.
